### PR TITLE
Recalculate "Content-Length"-Header after widget injection

### PR DESCRIPTION
**Pull request by trampi**
~~Unfortunately i did not found a way to remove an existing header (except creating a~~

It would be possible to override the header methods in HttpServletResponseBufferWrapper to prevent Content-Length headers from being set. Unfortunately, i realized this right now while writing this pull request.

This pull request however determines the size of the content in the response charset and sets the content length header accordingly.
I would understand if you favor the removal of the content length header. If so, decline this pull request. I will then rewrite this probably on friday.

### CODE REVIEW

**Comment by felixbarny** [comment]
Setting the content-length to -1 would probably do the same as omitting the header. Bug recalculating it might even be better. The only downside is, that `String#getBytes(String charset)` re-encodes the string creates a new char array. But I don't think this is a big performance issue.

**Comment by felixbarny** [comment]
The test fails on travis and on my machine. On travis the recalculated size is 5693, on my machine it is 5633 and on yours it apparently is 5534. Who is right ;) ?

**Comment by trampi** [comment]
This is indeed funny. I expected a regression regarding Linux vs Windows, but that your result also differs from travis ... OSX shouldn't behave THAT differently ;-)
I would have expected that the windows content length is bigger (CRLF vs. LF)
I admit it: In the current state its a poor test :smile:.

On friday i will do two new attempts:
* set content length to -1
* ignore all header-additions regarding Content-Length.

And i will have a look at WebSphere 8.5:
https://www.genuitec.com/products/myeclipse/download/websphere-developers/

**Comment by felixbarny** [comment]
I don't know about all the implications if the Content-Length is not set or negative, but since it's not possible to determine the content length for most dynamic content like jsps before the headers are sent to the client and the rest is still being generated anyway, I think this is not a problem.

TL;DR: Even (google.com)[http://www.google.com] doesn't sent a Content-Length header.

If we decide to take the recalculation path, we could take a look at how dandelion does that: https://github.com/dandelion/dandelion/blob/af4fd25431ba4dab9a86fb19574ff2121a4cec7a/dandelion-core/src/main/java/com/github/dandelion/core/web/handler/impl/AssetInjectionPostHandler.java#L185

**Comment by ponschab** [comment]
I woudn't spend any effort on calculating the response size. HTTP does not require sending a Content-Lenght header for the response.
http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.13:
| Applications SHOULD use this field to indicate the transfer-length of the message-body
Omitting the Content-Lenght header would be best, otherwise set it to -1.
